src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -49,6 +49,14 @@ export class Selection<T = any> {
             const trackBy: TrackByFunction<T> = this._items.trackBy;
             let selectionUpdated: boolean = false;
 
+            // if the currentSingle has been set before data was loaded, we look up and save the ref from current data set
+            if (this.currentSingle && !this.prevSingleSelectionRef) {
+              if (this._items.all && this._items.trackBy) {
+                const lookup = this._items.all.findIndex(maybe => maybe === this.currentSingle);
+                this.prevSingleSelectionRef = this._items.trackBy(lookup, this.currentSingle);
```

Comment:
I was about to ask you to change something, but then I realized this actually uses `trackBy` with index `-1` if we can't find the exact same reference as the pre-selected item in the overall list. That's perfect, it allows us to support proper trackBy (like by id) even for pre-selection.

Nothing to change here.